com.ibm.wala.cast.python.jython3/source/com/ibm/wala/cast/python/parser/PythonModuleParser.java

```diff
@@ -61,6 +72,85 @@ protected PythonParser<ModuleEntry>.CAstVisitor makeVisitor(
       WalkContext context, WalaPythonParser parser) {
     return new CAstVisitor(context, parser) {
 
+      @Override
+      public CAstNode visitImport(Import imp) throws Exception {
+        Optional<String> s =
+            imp.getInternalNames().stream()
+                .map(alias::getInternalName)
+                .reduce((a, b) -> a + "/" + b);
+
+        if (s.isPresent()) {
+          String moduleName = s.get().replace('.', '/');
+          LOGGER.finer("Module name from " + imp + " is: " + moduleName);
+
+          // if it's a package.
+          if (moduleName.indexOf('/') != -1) {
+            if (!isLocalModule(moduleName)) moduleName += "/" + MODULE_INITIALIZATION_ENTITY_NAME;
+
+            LOGGER.finer("Module name from " + imp + " is: " + moduleName);
+
+            if (isLocalModule(moduleName)) {
+              List<File> pythonPath = PythonModuleParser.this.getPythonPath();
+              LOGGER.info("PYTHONPATH is: " + pythonPath);
+
+              // If there is a PYTHONPATH specified.
+              if (pythonPath != null && !pythonPath.isEmpty()) {
+                // Adjust the module name per the PYTHONPATH.
+                Optional<SourceModule> localModule = getLocalModule(moduleName);
+
+                for (File pathEntry : pythonPath) {
+                  Path modulePath =
+                      localModule
+                          .map(SourceModule::getURL)
+                          .map(URL::getFile)
+                          .map(Path::of)
+                          .orElseThrow(IllegalStateException::new);
+                  LOGGER.finer("Found module path: " + modulePath);
+
+                  if (modulePath.startsWith(pathEntry.toPath())) {
+                    // Found it.
+                    Path scriptRelativePath = pathEntry.toPath().relativize(modulePath);
+                    LOGGER.finer("Relativized path is: " + scriptRelativePath);
+
+                    // Remove the file extension if it exists.
+                    moduleName = scriptRelativePath.toString().replaceFirst("\\.py$", "");
+
+                    // Use the beginning segment initialization file.
+                    moduleName = moduleName.split("/")[0] + "/" + MODULE_INITIALIZATION_ENTITY_NAME;
+
+                    LOGGER.fine("Using module name: " + moduleName);
+                    break;
+                  }
+                }
+              }
+
+              String yuck = moduleName;
```

Comment:
?

Reply:
That was in the original code :). My guess is that @juliandolby was frustrated with the requirement of lambdas to only accepted (effectively) final variables.

Reply:
Ok we can leave it :-)